### PR TITLE
[FLINK-13221][Table SQL / Planner][Release-1.9] Blink planner should set ScheduleMode to LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST for batch jobs

### PR DESCRIPTION
**Pull request by liyafan82**
## What is the purpose of the change

Make batch jobs use LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST as the default scheduling mode.

## Brief change log

  - Replace LAZY_FROM_SOURCES with LAZY_FROM_SOURCES_WITH_BATCH_SLOT_REQUEST for batch jobs.
  

## Verifying this change

This change is already covered by existing tests, such as *(please describe tests)*.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)
  - If yes, how is the feature documented? (not applicable)

### CODE REVIEW

File: flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java [hunk 1]
```diff
@@ -455,6 +455,15 @@ protected void restartTasks(final Set<ExecutionVertexID> verticesToRestart) {
 			this.delegate = checkNotNull(delegate);
 		}
 
+		@Override
+		public CompletableFuture<LogicalSlot> allocateBatchSlot(
```
Comment (KurtYoung) [inline flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java]: Is this necessary?

Comment (liyafan82) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java]: Yes. Without this method override, the default implementation of the super class will be called, which is based on SlotProvider#allocateBatchSlot. The default implementation throws an UnsupportedOperationException, which will cause some UT to fail.

Comment (KurtYoung) [reply flink-runtime/src/test/java/org/apache/flink/runtime/executiongraph/AdaptedRestartPipelinedRegionStrategyNGFailoverTest.java]: Thanks for the clarification!

**Comment by KurtYoung** [comment]
LGTM, waiting travis pass

**Pull request merged**